src/views/components/paymentRequestForm.js

```diff
@@ -0,0 +1,81 @@
+import React, { useState, useEffect } from 'react';
+import { Form, Button, Alert } from 'react-bootstrap';
+import DatePicker from 'react-datepicker';
+import "react-datepicker/dist/react-datepicker.css";
+import useCreatePayment from '../../controllers/useCreatePayment.js';
+
+function PaymentRequestForm() {
+    const [name, setName] = useState('');
+    const [billingAddress, setBillingAddress] = useState('');
+    const [CardNumber, setCardNumber] = useState('');
+    const [expiryDate, setExpiryDate] = useState(new Date());
+    const [CVC, setCVC] = useState('');
+
+    const {createRequest, payReq, error} = useCreatePayment()
+
+    const handleSubmit = async (event) => {
+        event.preventDefault();
+        const req = {
+            name, billingAddress, CardNumber:parseInt(CardNumber), expiryDate, CVC:parseInt(CVC) 
+        };
+        console.log(req)
+        await createRequest(req);
+    };
+
+    return (
+        <Form onSubmit={handleSubmit}>
+        
+            <Form.Group controlId="formName" required>
+                <Form.Label>Cardholder Name</Form.Label>
+                <Form.Control type="text" placeholder="Enter name" value={name} onChange={(e) => setName(e.target.value)} data-test-id="name-input" required/>
+            </Form.Group>
+
+            <Form.Group controlId="formAddress" required>
+                <Form.Label>Billing Address</Form.Label>
+                <Form.Control type="text" placeholder="Enter billing address" value={billingAddress} onChange={(e) => setBillingAddress(e.target.value)} data-test-id="address-input" required/>
+            </Form.Group>
+
+            <Form.Group controlId="formCardNumber" required>
+                <Form.Label>Card Number</Form.Label>
+                <Form.Control type="text" placeholder="Enter card number" value={CardNumber} onChange={(e) => setCardNumber(e.target.value)} data-test-id="card-number-input" required/>
+            </Form.Group>
+
+            <Form.Group controlId="formExpiryDate" required>
+                <Form.Label>Card Expiry Date</Form.Label>
+                <DatePicker.default 
```

Comment:
I think you can remove the time selection - bank cards only have the expiry date but not time afaik...